From 5c2a38beb7b26c153bfdd5667d1590e4f9bda4a5 Mon Sep 17 00:00:00 2001
From: Piotr Ziecik <kosmo@semihalf.com>
Date: Mon, 7 Apr 2014 15:29:18 +0200
Subject: [PATCH 1546/1825] alp: dfev: Enable Dragonite DFEV SLIC support.

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit b85daec3cf7bbf9ac701e9f0b755b957f0695570

	This commit enables compilation of the DFEV SLIC support when
	Dragonite driver is compiled-in.

Change-Id: I1a748a568a2d4fb2af91c834e15dd6ccdc9025de
Signed-off-by: Piotr Ziecik <kosmo@semihalf.com>
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/6995
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../plat-armada/mv_drivers_lsp/mv_phone/Makefile   |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/Makefile b/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/Makefile
index 708e251..a4d9847 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/Makefile
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_phone/Makefile
@@ -30,10 +30,18 @@ PHONE_OBJS += $(LSP_SPI_PATH)/spi_dev.o
 EXTRA_CFLAGS += -DMV_KERNEL_SLIC_SUPPORT
 
 # DFEV support
+IPC_DFEV_MODE = SMP
+ifeq ($(CONFIG_MV_IPC_LINUX_DRIVER),y)
+	IPC_DFEV_MODE = AMP
+endif
+ifeq ($(CONFIG_MV_IPC_FREERTOS_DRIVER),y)
+	IPC_DFEV_MODE = AMP
+endif
+
 ifeq ($(CONFIG_LANTIQ_DFEV_SLIC_SUPPORT),y)
 	PHONE_OBJS += $(LSP_SLIC_PATH)/sdd_if.o
 	PHONE_OBJS += $(LSP_SLIC_PATH)/dfev_if.o
-ifeq ($(CONFIG_MV_AMP_ENABLE),y)
+ifeq ($(IPC_DFEV_MODE),AMP)
 	PHONE_OBJS += $(LSP_DFEV_PATH)/ipc_dfev.o
 else
 	PHONE_OBJS += $(LSP_DFEV_PATH)/ipc_dfev_smp.o
-- 
1.7.5.4

